x-pack/plugins/ml/public/application/util/index_utils.ts

```diff
@@ -54,7 +51,15 @@ export interface DataViewAndSavedSearch {
   dataView: DataView | null;
 }
 
-export async function getDataViewAndSavedSearch(savedSearchId: string) {
+export async function getDataViewAndSavedSearch({
```

Comment:
❓ did you consider providing dependencies separately? e.g. 
```ts
export const getDataViewAndSavedSearchCallback = (deps: {
    savedSearchService: SavedSearchPublicPluginStart;
    dataViewsService: DataViewsContract;
}) => (savedSearchId: string) => {
    // ...
}
```

It could be our own `ml` service with more methods as well.